node/core/candidate-validation/src/lib.rs

```diff
@@ -163,50 +167,68 @@ where
 					pov,
 					response_sender,
 				) => {
-					let _timer = metrics.time_validate_from_exhaustive();
-
-					let res = validate_candidate_exhaustive(
-						&mut validation_host,
-						persisted_validation_data,
-						validation_code,
-						descriptor,
-						pov,
-						&metrics,
-					)
-					.await;
-
-					match res {
-						Ok(x) => {
-							metrics.on_validation_event(&x);
-
-							if let Err(_e) = response_sender.send(x) {
-								tracing::warn!(
-									target: LOG_TARGET,
-									"Requester of candidate validation dropped",
-								)
-							}
-						},
-						Err(e) => return Err(e),
-					}
+					let bg = {
+						let metrics = metrics.clone();
+						let validation_host = validation_host.clone();
+
+						async move {
+							let _timer = metrics.time_validate_from_exhaustive();
+							let res = validate_candidate_exhaustive(
+								validation_host,
+								persisted_validation_data,
+								validation_code,
+								descriptor,
+								pov,
+								&metrics,
+							)
+							.await;
+
+							metrics.on_validation_event(&res);
+							let _ = response_sender.send(res);
+						}
+					};
+
+					ctx.spawn("validate-from-exhaustive", bg.boxed())?;
 				},
 			},
 		}
 	}
 }
 
-async fn runtime_api_request<T, Context>(
-	ctx: &mut Context,
+struct RuntimeRequestFailed;
+
+async fn runtime_api_request<T, Sender>(
+	sender: &mut Sender,
 	relay_parent: Hash,
 	request: RuntimeApiRequest,
 	receiver: oneshot::Receiver<Result<T, RuntimeApiError>>,
-) -> SubsystemResult<Result<T, RuntimeApiError>>
+) -> Result<T, RuntimeRequestFailed>
 where
-	Context: SubsystemContext<Message = CandidateValidationMessage>,
-	Context: overseer::SubsystemContext<Message = CandidateValidationMessage>,
+	Sender: SubsystemSender,
 {
-	ctx.send_message(RuntimeApiMessage::Request(relay_parent, request)).await;
+	sender
+		.send_message(RuntimeApiMessage::Request(relay_parent, request).into())
+		.await;
 
-	receiver.await.map_err(Into::into)
+	receiver
+		.await
+		.map_err(|_| {
+			tracing::debug!(target: LOG_TARGET, ?relay_parent, "Runtime API request dropped");
+
+			RuntimeRequestFailed
+		})
+		.and_then(|res| {
+			res.map_err(|e| {
+				tracing::debug!(
```

Comment:
What's the reasoning behind making those two `debug!`?

Reply:
They are unexpected but non-fatal - trace! is too low, but warn! is too high.